inline_scan.sh

```diff
@@ -346,16 +336,40 @@ start_analysis() {
 	    fi
 	    exit 1
 	fi
+}
 
-    FULLTAG="${SCAN_IMAGES[0]}"
-    check_status_with_digest
+# This is done instead of the -g option, as we want to tie the RepoDigest value present in the image
+# with the image id as much as possible, instead of generating our own digest or via skopeo.
+get_repo_digest_id() {
+    # Check to see if repo digest exists
+    if [[ "${SYSDIG_IMAGE_DIGEST}" == 'sha256:123456890abcdefg' ]]; then
+        DIGESTS=$(docker inspect --format="{{.RepoDigests}}" "${SCAN_IMAGES[0]}")
+    fi
+
+    BASE_IMAGE=$(echo ${IMAGE_NAMES[0]} | cut -d / -f 2 | cut -d : -f 1)
+
+     if [[ ${digest} == *"${BASE_IMAGE}"* ]]; then
+        DIGEST=$(echo ${digest} | tr -d '[' | tr -d ']' | cut -d : -f 2 | cut -d '"' -f 1)
+        SYSDIG_IMAGE_DIGEST=$(echo "sha256:${DIGEST}")
```

Comment:
Typo: `${digest}` should be `${DIGESTS}`

And this still doesn't work when we have more than on digests:

```+ [[ [alpine@sha256:c19173c5ada610a5989151111163d28a67368362762534d8a8121ce95cf2bd5a alpine@sha256:e4355b66995c96b4b468159fc5c7e3540fcef961189ca13fee877798649f531a] == *\a\l\p\i\n\e* ]]
++ echo '[alpine@sha256:c19173c5ada610a5989151111163d28a67368362762534d8a8121ce95cf2bd5a' 'alpine@sha256:e4355b66995c96b4b468159fc5c7e3540fcef961189ca13fee877798649f531a]'
++ tr -d '['
++ tr -d ']'
++ cut -d : -f 2
++ cut -d '"' -f 1
+ DIGEST='c19173c5ada610a5989151111163d28a67368362762534d8a8121ce95cf2bd5a alpine@sha256'
```

Reply:
I tested this with multiple digests, with our alpine:latest actually. Are you sure your test input is in the right format ?

Reply:
I had to fix the unbound variable error to get there. Maybe you forgot to push some code?

Reply:
What's the test input that triggered that error ? How can i reproduce that ?

Reply:
No, I see it now. I will update this